### PR TITLE
Remove default flags. Disable mtd builds. Clean up logging.c 

### PR DESCRIPTION
**Pull request by nikita-s-wrk**

### CODE REVIEW

File: script/build [hunk 1]
```diff
@@ -38,6 +38,26 @@ readonly OT_OPTIONS=(
     "-DCMAKE_BUILD_TYPE=MinSizeRel"
     "-DOT_PLATFORM=external"
     "-DOT_SLAAC=ON"
+    "-DOT_COMMISSIONER=ON"
```
Comment (jwhui) [inline script/build:41]: nit: sort in alphabetical ordering?

Comment (nikita-s-wrk) [reply script/build]: Done

File: script/build [hunk 2]
```diff
@@ -36,10 +36,32 @@ readonly QORVO_PLATFORMS=(gp712 qpg6095 qpg6100 qpg7015m)
...
+    "-DOT_ECDSA=ON"
+    "-DOT_LOG_LEVEL_DYNAMIC=ON"
+    "-DOT_MAC_FILTER=ON"
+    "-DOT_MTD_NETDIAG=ON"
+    "-DOT_SERVICE=ON"
     "-DOT_SLAAC=ON"
+    "-DOT_SNTP_CLIENT=ON"
 )
 
+
```
Comment (jwhui) [inline script/build:64]:
```suggestion

```

File: script/build [hunk 3]
```diff
@@ -36,10 +36,33 @@ readonly QORVO_PLATFORMS=(gp712 qpg6095 qpg6100 qpg7015m)
...
+    "-DOT_DNS_CLIENT=ON"
+    "-DOT_ECDSA=ON"
+    "-DOT_LOG_LEVEL_DYNAMIC=ON"
+    "-DOT_MAC_FILTER=ON"
+    "-DOT_MTD_NETDIAG=ON"
+    "-DOT_SERVICE=ON"
+    "-DOT_SNTP_CLIENT=ON"
+)
+
+
```
Comment (jwhui) [inline script/build:65]:
```suggestion

```

**Comment by jwhui** [comment]
@nikita-s-wrk , we recently just did the opposite in another repo - see https://github.com/openthread/ot-nrf528xx/issues/41 for the discussion. It would be nice to keep the platform-specific repos consistent. Thoughts?

**Comment by nikita-s-wrk** [comment]
@jwhui For embedded platforms we would like to keep bare minimum enabled(Joining, SLAAC, COAP) to better keep track if we still fit in flash while openthread codebase grows. 
There are per-platform flags definition in openthread's check-size script however. What do you think on moving flags to platform repos and add script/build parameter to build specific ones? I.e. `script/build qpg6095 default` and `script/build qpg6095 full`.
That will eliminate duplicating flag definitions across CI scripts for every platform/script and let vendors control platform flags  without touching main repo.

**Comment by jwhui** [comment]
@nikita-s-wrk , can we add the flags to [script/test](https://github.com/openthread/ot-qorvo/blob/main/script/test) instead?

**Comment by jwhui** [comment]
> @nikita-s-wrk , can we add the flags to [script/test](https://github.com/openthread/ot-qorvo/blob/main/script/test) instead?

@nikita-s-wrk , thoughts on this?

**Comment by nikita-s-wrk** [comment]
@jwhui Removed default flags, please take a look. 
We have also disabled mtd builds for now, this will be addressed in separate issue.

**Comment by jwhui** [comment]
> @jwhui Removed default flags, please take a look.
> We have also disabled mtd builds for now, this will be addressed in separate issue.

Thanks @nikita-s-wrk . Is the title of this PR still accurate? If not, can you update the title of this PR and add a description?

**Comment by jwhui** [comment]
@nikita-s-wrk , it appears the build checks are failing. Can you take a look?

**Comment by nikita-s-wrk** [comment]
@jwhui Seems it was sporadic failure of github actions. I've force pushed with new commit id to retrigger github actions.

**Comment by jwhui** [comment]
> @jwhui Seems it was sporadic failure of github actions. I've force pushed with new commit id to retrigger github actions.

@nikita-s-wrk , the CI checks are still failing:

```
+ ninja ot-cli-ftd
ninja: error: '../third_party/Qorvo/repo/qpg6095/lib/libmbedtls_alt.a', needed by 'bin/ot-cli-ftd', missing and no known rule to make it
Error: Process completed with exit code 1.
```

**Review by jwhui — APPROVED** [review]

**Pull request merged**